src/com/stanimiriliev/chaptersix/pagebean/PageBean.java

```diff
@@ -0,0 +1,143 @@
+package com.stanimiriliev.chaptersix.pagebean;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class receive list and divide it to pages
+ * @author Stanimir Iliev
+ * @since 24.07.2017
+ */
+public class PageBean <T>{
+
+    private final List<T> src;
+    public final int pageSize;
+    private int upIndex = -1;
+    private int downIndex;
+    private int pageIndex = -1;
+    private boolean hasNext = true, hasPrevious = false;
+    private List<T> list;
+
+    public PageBean(List<T> src, int pageSize){
+        if(src == null || src.size() == 0 || pageSize <= 0)
+            throw new IllegalArgumentException("This error may occur because of several things:\n" +
+                    "Source list is null\nSource list is empty\nPage size is less than or equal to zero");
+        this.src = src;
+        this.pageSize = pageSize;
+    }
+
+    /**
+     * Check if the downIndex or upIndex has reached the its limits
+     * and updates hasPrevious and hasNext variables
+     */
+    private void check(){
+        if(downIndex == 0)
+            hasPrevious = false;
+        if(upIndex == src.size() - 1)
+            hasNext = false;
+        if(upIndex < src.size() - 1 && downIndex > 0){
+            hasNext = true;
+            hasPrevious = true;
+        }
+    }
+
+    /**
+     * Reverses order of the elements of the list
+     * @param list -> The list which will be reversed
+     * @return reversed list
+     */
+    private List<T> reverseList(List<T> list){
+        List<T> temp = new ArrayList<>();
+        for(int i = list.size() - 1; i >= 0; i--)
+            temp.add(list.get(i));
+        return temp;
```

Comment:
In such cases you can use helper libraries like google guava 
https://google.github.io/guava/releases/22.0/api/docs/com/google/common/collect/Lists.html
```
Lists.reverse(list);
```